components/org.wso2.carbon.identity.governance/src/main/java/org/wso2/carbon/identity/governance/listener/IdentityMgtEventListener.java

```diff
@@ -80,12 +80,13 @@ public boolean doPreAuthenticate(String userName, Object credential,
             log.debug("Pre authenticator is called in IdentityMgtEventListener");
         }
         IdentityUtil.clearIdentityErrorMsg();
-        IdentityUtil.threadLocalProperties.get().remove(RE_CAPTCHA_USER_DOMAIN);
 
         // This is used set domain of the user when authentication is failed for an existing user. This is required
         // for re-captcha feature.
-        IdentityUtil.threadLocalProperties.get().put(RE_CAPTCHA_USER_DOMAIN,
-                IdentityGovernanceUtil.getUserStoreDomainName(userStoreManager));
+        if (isUserExistsInDomain(userStoreManager, userName)) {
```

Comment:
@thanujalk : Do we really need to do this check here and set this thread local at this point ? IIRC we avoided checking isExisting user in multiple places due to perf issues. This introduces another is existing method. (We are already doing this isExisting check in doPostAuthenticate method)